physx/src/foundation.rs

```diff
@@ -92,6 +94,25 @@ pub trait Foundation: Class<physx_sys::PxFoundation> + Sized {
         }
     }
 
+    /// Tries to create a PxFoundation with the provided allocator and error callbacks.
+    /// `error_callback` must live as long as the returned `Foundation`
```

Comment:
Is it ok to just document these things like this?
Or is there backwards compatible way to enforce it?

Reply:
yes any `unsafe` function should ideally have a doc line that describes the function and then a `# Safety` section where the unsafe invariants are described like this. There is even a lint for that that we ideally should use in this repo (and do use in ark)

Reply:
example: https://rust-lang.github.io/api-guidelines/documentation.html#c-failure

Reply:
Thanks.
I added safety sections